gammapy/cube/fov.py

```diff
@@ -0,0 +1,97 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""FoV background estimation."""
+import logging
+from gammapy.maps import Map
+from gammapy.modeling import Fit
+
+__all__ = ["FoVBackgroundMaker"]
+
+log = logging.getLogger(__name__)
+
+
+class FoVBackgroundMaker:
+    """Normalize template background on the whole field-of-view.
+
+    The dataset background model can be simply scaled (method="scale") or fitted (method="fit")
+    on the dataset counts.
+
+    The normalization is performed outside the exclusion mask taht is passed on init.
+
+    Parameters
+    ----------
+    method : str in ['fit', 'scale']
+        the normalization method to be applied. Default 'fit'.
+    exclusion_mask : `~gammapy.maps.WcsNDMap`
+        Exclusion mask
+    """
+
+    def __init__(self, method="fit", exclusion_mask=None):
+        if method in ["fit", "scale"]:
+            self.method = method
+        else:
+            raise ValueError(
+                f"Incorrect method for FoVBackgroundMaker: {method}."
+            )
+        self.exclusion_mask = exclusion_mask
+
+    def run(self, dataset):
+        """Run FoV background maker.
+
+        Fit the background model norm
+
+        Parameters
+        ----------
+        dataset : `~gammapy.cube.fit.MapDataset`
+            Input map dataset.
+
+        """
+        mask_fit = dataset.mask_fit
+        dataset.mask_fit = self._reproject_exclusion_mask(dataset)
+
+        if self.method is "fit":
+            self._fit_bkg(dataset)
+        else:
+            self._scale_bkg(dataset)
+
+        dataset.mask_fit = mask_fit
+        return dataset
+
+    def _reproject_exclusion_mask(self, dataset):
+        """Reproject the exclusion on the dataset geometry"""
+        mask_map = Map.from_geom(dataset.counts.geom)
+        if self.exclusion_mask is not None:
+            coords = dataset.counts.geom.get_coord()
+            vals = self.exclusion_mask.get_by_coord(coords)
+            mask_map.data += vals
+
+        return mask_map.data.astype("bool")
+
+    def _fit_bkg(self, dataset):
+        """Fit the FoV background model on the dataset counts data"""
+
+        # Here we assume that the model is only the background model
+        # TODO : freeze all model components not related to background model?
```

Comment:
I think freezing all components instead of the background is probably the better (safer) default behaviour, especially if the default method is `fit` as well.

Reply:
That's right. 
I changed the default to `scale` which is probably more natural.
I added the parameters freezing in case a `SkyModel` is set on the dataset. Parameters are then unfrozen. A test is added to check for this.